### DISCUSSION TITLE
Is vue3 support going to be backward compatible in terms of component api's

### DISCUSSION THREAD

**Discussion by yenniSuresh** [vuetifyjs/vuetify#13634] [Q&A] [answered]
### Problem to solve
Hi, in my project, we are planning to go forward with vuetify with vue2 version, if possible I wanted to know how drastically vuetify's vue3 support will differ to vue2. If the migration effort from vuetify's vue2 to vue3 going to be low, medium or high? if the the components api's are going to be changed?

Thanks for your time, your answer will help me take good decision for my project. 

### Proposed solution
I am assuming, backward compatibility i.e, less changes to component api's will be correct way which will help existing vuetify users to upgrade easily

**Comment by KaelWD** [comment, marked as answer]
> If the migration effort from vuetify's vue2 to vue3 going to be low, medium or high?

low-medium, maybe high if you're relying on internals. We're aiming to keep the public JS API (props, slots, events) fairly close to 2.x, but with inconsistencies fixed and some new features added. Most of these changes will hopefully be able to be fixed automatically with codemods or eslint. Styles have some significant changes though so custom overrides may need to be rewritten, and components using internal APIs (like if you're extending our components, using our mixins, or modifying values with refs) will probably completely break as these are no longer available.

**Reply by yenniSuresh** [reply]
Thanks @KaelWD
